packages/core/types/src/types/core/attributes/blocks.ts

```diff
@@ -1,5 +1,75 @@
 import type { Attribute } from '..';
 
+interface TextInlineNode {
+  type: 'text';
+  text: string;
+  bold?: boolean;
+  italic?: boolean;
+  underline?: boolean;
+  strikethrough?: boolean;
+  code?: boolean;
+}
+
+interface LinkInlineNode {
+  type: 'link';
+  url: string;
+  children: TextInlineNode[];
+}
+
+interface ListItemInlineNode {
+  type: 'list-item';
+  children: DefaultInlineNode[];
+}
+
+type DefaultInlineNode = TextInlineNode | LinkInlineNode;
+type NonTextInlineNode = Exclude<DefaultInlineNode | ListItemInlineNode, TextInlineNode>;
+
+interface ParagraphBlockNode {
+  type: 'paragraph';
+  children: DefaultInlineNode[];
+}
+
+interface QuoteBlockNode {
+  type: 'quote';
+  children: DefaultInlineNode[];
+}
+
+interface CodeBlockNode {
+  type: 'code';
+  children: DefaultInlineNode[];
+}
+
+interface HeadingBlockNode {
+  type: 'heading';
+  level: 1 | 2 | 3 | 4 | 5 | 6;
+  children: DefaultInlineNode[];
+}
+
+interface ListBlockNode {
+  type: 'list';
+  format: 'ordered' | 'unordered';
+  children: (ListItemInlineNode | ListBlockNode)[];
+}
+
+interface ImageBlockNode {
+  type: 'image';
+  image: Attribute.GetValue<{
+    type: 'media';
+    allowedTypes: ['images'];
+    multiple: false;
+  }>;
+  children: [{ type: 'text'; text: '' }];
```

Comment:
what is this supposed to represent ? caption ?

Reply:
It's a silly slate constraint that we haven't managed to get rid of, it expects all nodes to have children but we don't want it for images

Reply:
I would reco to create a base Node Interface that they can all extend with children: unknown or any depending on your need

Reply:
Just wondering what's the value in extending a BaseNode if all nodes define type and children?